internal/controllers/operator_controller.go

```diff
@@ -357,6 +367,19 @@ func isBundleDepStale(bd *rukpakv1alpha1.BundleDeployment) bool {
 	return bd != nil && bd.Status.ObservedGeneration != bd.GetGeneration()
 }
 
+// mapBundleMediaTypeToBundleProvisioner maps an olm.bundle.mediatype property to a
+// rukpak bundle provisioner class name that is capable of unpacking the bundle type
+func mapBundleMediaTypeToBundleProvisioner(mediaType string) (string, error) {
+	switch mediaType {
+	case entity.MediaTypePlain:
+		return "core-rukpak-io-plain", nil
+	case entity.MediaTypeRegistry, "":
+		return "core-rukpak-io-registry", nil
```

Comment:
It's probably a good idea to make a comment about the empty string and explain why `""` means registry+plain.

Reply:
Comment added in https://github.com/operator-framework/operator-controller/pull/242/commits/012fa1475eb4819386fa2f00e0e98222a1f06a50